Explorer/Assets/Plugins/TexturesFuse/TexturesServerWrap/Unzips/ITexturesUnzip.cs

```diff
@@ -0,0 +1,185 @@
+using Cysharp.Threading.Tasks;
+using DCL.Diagnostics;
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility.Types;
+
+namespace Plugins.TexturesFuse.TexturesServerWrap.Unzips
+{
+    /// <summary>
+    ///     Provides decoded and compressed textures from raw memory.
+    ///     Its implementations don't guarantee thread safety.
+    /// </summary>
+    public interface ITexturesUnzip : IDisposable
+    {
+        /// <param name="bytes">Pointer to the array of encoded data, client guarantees the pointer will be valid for the whole duration of Task.</param>
+        /// <param name="bytesLength">Length of encoded data.</param>
+        /// <param name="type">Desired type that result will be consumed.</param>
+        /// <param name="token">Cancellation Token to cancel operation.</param>
+        UniTask<EnumResult<OwnedTexture2D, NativeMethods.ImageResult>> TextureFromBytesAsync(
+            IntPtr bytes,
+            int bytesLength,
+            TextureType type,
+            CancellationToken token
+        );
+
+        interface IOptions
+        {
+            Mode Mode { get; }
+
+            NativeMethods.Swizzle Swizzle { get; }
+
+            int MaxSide { get; }
+
+            NativeMethods.Adjustments Adjustments { get; }
+
+            class Const : IOptions
+            {
+                public Const(Mode mode, NativeMethods.Swizzle swizzle, int maxSide, NativeMethods.Adjustments adjustments)
+                {
+                    Mode = mode;
+                    Swizzle = swizzle;
+                    MaxSide = maxSide;
+                    Adjustments = adjustments;
+                }
+
+                public Mode Mode { get; }
+                public NativeMethods.Swizzle Swizzle { get; }
+                public int MaxSide { get; }
+                public NativeMethods.Adjustments Adjustments { get; }
+            }
+        }
+
+        public static ITexturesUnzip NewDefault()
+        {
+            var init = NativeMethods.InitOptions.NewDefault();
+            var options = new IOptions.Const(Mode.ASTC_6x6, NativeMethods.Swizzle.NewDefault(), 1024, NativeMethods.Adjustments.NewEmpty());
+            var index = 0;
+
+            return new PooledTexturesUnzip(
+                () => new TexturesUnzip(init, options, true)
+                   .WithLog((++index).ToString())
+
+                // .WithSemaphore() -
+                // is not required since PooledTexturesUnzip has synchronization for the access and prevents double calling of TextureFromBytesAsync
+               ,
+                3
```

Comment:
I would bind this number to the CPU cores

Reply:
Good point, and I'd to use (cores_count - 1) or (cores_count / 2) to avoid a competition with main thread